# STS score prediction Rest API

Created Rest API using sts_vx.ckpt.3
- GET: score estimation between two sentence (0-5)
- POST: score estimation on one sentence against multiple sentences (csv file with header 'sentence')

In [ ]:
!pip install transformers
!pip install Flask
import requests
import pandas as pd

In [2]:
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


In [72]:
example = pd.read_csv('/content/drive/MyDrive/data/KorNLUDatasets/KorSTS/sts_sentence.csv')
example.head()

Unnamed: 0                 sentence
0           0     한 소녀가 머리를 스타일링하고 있다.
1           1  한 무리의 남자들이 해변에서 축구를 한다.
2           2  한 여성이 다른 여성의 발목을 재고 있다.
3           3        한 남자가 오이를 자르고 있다.
4           4       한 남자가 하프를 연주하고 있다.

In [107]:
# predict: get/post   sentence_one, sentence_two / base_sen, uploaded_file

get_example = requests.get("http://127.0.0.1:5000/predict", 
                    {'sentence_one': '한 남자가 밧줄을 타고 올라간다.', 
                     'sentence_two': '한 남자가 밧줄을 타고 올라가고 있다.'})

op = open('/content/drive/MyDrive/data/KorNLUDatasets/KorSTS/sts_sentence.csv', 'r')
post_example = requests.post("http://127.0.0.1:5000/predict", 
                             params={'base_sen': '한 남자가 식당에서 밥을 먹고 있다.'}, 
                             files={'uploaded_file': op})


In [110]:
print(f'Similarity Score: {get_example.content}')

Similarity Score: b'4.906567573547363'


In [102]:
import ast
final = ast.literal_eval(post_example.content.decode('utf-8'))

In [100]:
top_ten = sorted(list(final.items()), key = lambda x : float(x[1]), reverse=True)[:10]

In [101]:
original = example['sentence'].tolist()
for index, score in top_ten:
    print(original[int(index)])

한 남자가 음식을 먹고 있다.
남자가 먹고 있다.
한 남자와 여자가 식당의 테이블에 앉아 있다.
식당 식탁에 둘러앉은 사람들 무리.
한 무리의 사람들이 이탈리아 식당의 테이블에 앉아 있다.
한 남자가 냄비에 쌀을 붓는다.
네 명의 아이들이 식당에서 의자에 앉아 있다.
두 명의 근로자가 파이프 옆에 앉아 점심을 먹고 있다
한 남자가 어느 날 밤늦게 만화를 보면서 팝콘을 먹는다.
한 남자가 쌀 한 송이를 팬에 붓고 있다.


## GET
### Sentence One
한 남자가 밧줄을 타고 올라간다.
### Sentence Two
한 남자가 밧줄을 타고 올라가고 있다.

### Score
4.906567573547363

.


## POST
### Base Sentence

한 남자가 식당에서 밥을 먹고 있다.

### Top Similar Sentences

한 남자가 음식을 먹고 있다.

남자가 먹고 있다.

한 남자와 여자가 식당의 테이블에 앉아 있다.

식당 식탁에 둘러앉은 사람들 무리.

한 무리의 사람들이 이탈리아 식당의 테이블에 앉아 있다.

한 남자가 냄비에 쌀을 붓는다.

네 명의 아이들이 식당에서 의자에 앉아 있다.

두 명의 근로자가 파이프 옆에 앉아 점심을 먹고 있다

한 남자가 어느 날 밤늦게 만화를 보면서 팝콘을 먹는다.

한 남자가 쌀 한 송이를 팬에 붓고 있다.
